server/routes_config.go

```diff
@@ -0,0 +1,173 @@
+}
+
+func (r *routesConfigImpl) ReadRoutesConfig(routesConfig string) error {
+	r.fileName = routesConfig
+
+	logrus.WithField("routesConfig", r.fileName).Info("Loading routes config file")
+
+	config, readErr := r.readRoutesConfigFile()
+
+	if readErr != nil {
+		if errors.Is(readErr, fs.ErrNotExist) {
+			logrus.WithField("routesConfig", r.fileName).Info("Routes config file doses not exist, skipping reading it")
+			// File doesn't exist -> ignore it
+			return nil
+		}
+		return errors.Wrap(readErr, "Could not load the routes config file")
+	}
+
+	Routes.RegisterAll(config.Mappings)
+	Routes.SetDefaultRoute(config.DefaultServer)
+	return nil
+}
+
+func (r *routesConfigImpl) AddMapping(serverAddress string, backend string) {
+	if !r.isRoutesConfigEnabled() {
+		return
+	}
+
+	config, readErr := r.readRoutesConfigFile()
+	if readErr != nil && !errors.Is(readErr, fs.ErrNotExist) {
+		logrus.WithError(readErr).Error("Could not read the routes config file")
+		return
+	}
+	if config.Mappings == nil {
+		config.Mappings = make(map[string]string)
+	}
+
+	config.Mappings[serverAddress] = backend
+
+	writeErr := r.writeRoutesConfigFile(config)
+	if writeErr != nil {
+		logrus.WithError(writeErr).Error("Could not write to the routes config file")
+		return
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"serverAddress": serverAddress,
+		"backend":       backend,
+	}).Info("Added route to routes config")
+
+	return
+}
+
+func (r *routesConfigImpl) SetDefaultRoute(backend string) {
+	if !r.isRoutesConfigEnabled() {
+		return
+	}
+
+	config, readErr := r.readRoutesConfigFile()
+	if readErr != nil && !errors.Is(readErr, fs.ErrNotExist) {
+		logrus.WithError(readErr).Error("Could not read the routes config file")
+		return
+	}
+
+	config.DefaultServer = backend
+
+	writeErr := r.writeRoutesConfigFile(config)
+	if writeErr != nil {
+		logrus.WithError(writeErr).Error("Could not write to the routes config file")
+		return
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"backend": backend,
+	}).Info("Set default route in routes config")
+
+	return
+}
+
+func (r *routesConfigImpl) DeleteMapping(serverAddress string) {
+	if !r.isRoutesConfigEnabled() {
+		return
+	}
+
+	config, readErr := r.readRoutesConfigFile()
+	if readErr != nil && !errors.Is(readErr, fs.ErrNotExist) {
+		logrus.WithError(readErr).Error("Could not read the routes config file")
+		return
+	}
+
+	delete(config.Mappings, serverAddress)
+
+	writeErr := r.writeRoutesConfigFile(config)
+	if writeErr != nil {
+		logrus.WithError(writeErr).Error("Could not write to the routes config file")
+		return
+	}
+
+	logrus.WithField("serverAddress", serverAddress).Info("Deleted route in routes config")
+
+	return
+}
+
+func (r *routesConfigImpl) isRoutesConfigEnabled() bool {
+	return r.fileName != ""
+}
+
+func (r *routesConfigImpl) readRoutesConfigFile() (routesConfigStructure, error) {
+	r.RLock()
+	defer r.RUnlock()
+
+	config := routesConfigStructure{
+		"",
+		make(map[string]string),
+	}
```

Comment:
I don't believe it is necessary to prep the empty content. I think just a `var config routesConfigStucture` would suffice.